src/Enzo/enzo_EnzoMethodM1Closure.hpp

```diff
@@ -0,0 +1,158 @@
+  const std::string sigE_string(int i, int j) throw()
+    {return "sigE_" + std::to_string(i) + std::to_string(j);}
+  const std::string eps_string (int i) throw()
+    {return "eps_"  + std::to_string(i);}
+  const std::string mL_string  (int i) throw()
+    {return "mL_" + std::to_string(i);}
+
+  
+  //--------- INJECTION STEP -------
+
+  double get_star_temperature(double M) throw();
+
+  double get_radiation_custom(EnzoBlock * enzo_block,
+             double energy, double pmass, double plum, 
+             double dt, double inv_vol, int i, int igroup) throw();
+
+  double get_radiation_blackbody(EnzoBlock * enzo_block, double pmass, 
+             double freq_lower, double freq_upper, double clight, 
+             double dt, double cell_volume, int i, int igroup) throw();
+
+  void inject_photons(EnzoBlock * enzo_block, int igroup) throw();
+
+  //--------- TRANSPORT STEP --------
+
+
+  double flux_function (double U_l, double U_lplus1,
+                        double Q_l, double Q_lplus1,
+                        double clight, double lmin, double lmax, std::string type) throw();
+
+  double compute_hll_eigenvalues(double f, double theta, double * lmin, double * lmax, double clight) throw();
+
+  double deltaQ_faces (double U_l, double U_lplus1, double U_lminus1, 
+                       double Q_l, double Q_lplus1, double Q_lminus1,
+                       double clight, double lmin, double lmax, std::string flux_type) throw();
+
+  void get_reduced_variables (double * chi_idx, double (*n_idx)[3], int i, double clight,
+                              enzo_float * N, enzo_float * Fx, enzo_float * Fy, enzo_float * Fz) 
+                              throw(); 
+
+  void get_pressure_tensor (EnzoBlock * enzo_block, 
+                       enzo_float * N, enzo_float * Fx, enzo_float * Fy, enzo_float * Fz,
+                       double clight) throw();
+
+  void get_U_update (EnzoBlock * enzo_block, double * N_update, 
+                       double * Fx_update, double * Fy_update, double * Fz_update, 
+                       enzo_float * N, enzo_float * Fx, enzo_float * Fy, enzo_float * Fz,
+                       double hx, double hy, double hz, double dt, double clight, 
+                       int i, int idx, int idy, int idz) throw();
+
+  void solve_transport_eqn (EnzoBlock * enzo_block, int igroup) throw();
+
+  void add_LWB (EnzoBlock * enzo_block, double J21);
+
+  //---------- THERMOCHEMISTRY STEP ------------
+
+
+  // Interaction with matter is completely local, so don't need a refresh before this step
+  void D_add_attenuation ( EnzoBlock * enzo_block, double * D, double clight, int i, int igroup) throw(); 
+
+  double get_alpha (double T, int species, char rec_case) throw();
+
+  int get_b_boolean (double E_lower, double E_upper, int species) throw();
+
+  void C_add_recombination (EnzoBlock * enzo_block, double * C, 
+                              enzo_float * T, int i, int igroup, double E_lower, double E_uppe) throw();
```

Comment:
Is there any reason this function doesn't just return `C`? That's generally easier for the compiler to understand.

Reply:
Not particularly. Changed function to return C instead